link_linux.go

```
@@ -1580,9 +1586,57 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 	}
 	*link.Attrs() = base
 
+	// If the tuntap attributes are not updated by netlink due to
+	// an older driver, use sysfs
+	if link != nil && linkType == "tun" {
+		tuntap := link.(*Tuntap)
+
+		if tuntap.Mode == 0 {
+			ifname := tuntap.Attrs().Name
+			if flags, err := readSysPropAsInt64(ifname, "tun_flags"); err == nil {
+
+				if (flags & unix.IFF_TUN) != 0 {
+					tuntap.Mode = unix.IFF_TUN
+				} else if (flags & unix.IFF_TAP) != 0 {
+					tuntap.Mode = unix.IFF_TAP
+				}
+
+				if flags&unix.IFF_PERSIST == 0 {
+					tuntap.NonPersist = true
+				} else {
+					tuntap.NonPersist = false
+				}
+
+			}
+			if owner, err := readSysPropAsInt64(ifname, "owner"); err == nil && owner > 0 {
+				tuntap.Owner = uint32(owner)
+			}
+			if group, err := readSysPropAsInt64(ifname, "group"); err == nil && group > 0 {
```

We should not discard the error.

---

best effort logic as explained by @amshinde